db/system_keyspace.cc

```diff
@@ -1997,6 +2036,60 @@ future<int> increment_and_get_generation() {
     });
 }
 
+mutation make_client_mutation(client_data&& cd) {
+    auto&& schema = db::system_keyspace::clients();
+    auto timestamp = api::new_timestamp();
+    auto pk = partition_key::from_single_value(*schema, inet_addr_type->decompose(cd.ip));
+
+    mutation m_to_apply{schema, pk};
+    const auto ck = clustering_key_prefix(std::vector<bytes>{int32_type->decompose(cd.port)});
+
+    sstring connection_stage_str;
+    switch (cd.connection_stage) {
+    case service::client_state::auth_state::UNINITIALIZED:
+        connection_stage_str = "established";  // This is C* convention
+        break;
+    case service::client_state::auth_state::AUTHENTICATION:
+        connection_stage_str = "authenticating";
+        break;
+    case service::client_state::auth_state::READY:
+        connection_stage_str = "ready";
+        break;
+    default:
+        connection_stage_str = to_sstring((uint8_t)cd.connection_stage);
+    }
```

Comment:
Wouldn't a `static const std::map<auth_state, sstring>` or sth like that be more elegant?
But I guess this is just a matter of preference.

Reply:
In any case, I'd suggest to move this code to a `make_connection_stage_string` or something like that.

Reply:
A case-switch is more efficient.

Reply:
I'm sticking to `switch`, but putting this into another function is a good idea.